### ISSUE TITLE
How to make a diagram fit to page

### ISSUE THREAD

**Issue by purpl3F0x** [Jollywatt/typst-fletcher#58]
Hello, how can i make a diagram auto-fit inside the page area ?

**Comment by Jollywatt** [comment]
Not sure, it isn't supported directly as an option. But maybe we could add `width/height` options to `diagram()` as an alternative to specifying the cell sizes, and these could be made to take `auto`.

**Comment by JL710** [comment]
I think it would be cool if there was a scale option. Cetz itself provides one, but when using it in the render function that breaks the result. There is also auto-scale on content that could make this easier.

**Comment by Jollywatt** [comment]
You could just wrap the diagram in `scale()`?

**Comment by JL710** [comment]
Yes, that wold make he diagram appear smaller, but it would take the same space.

Example:
```
#box(fill: red, scale(20%)[
  #import "@preview/fletcher:0.5.5" as fletcher: diagram, node, edge
  // You can specify nodes in math-mode, separated by `&`:
  #diagram($
  G edge(f, ->) edge("d", pi, ->>) & im(f) \
  G slash ker(f) edge("ur", tilde(f), "hook-->")
  $)
])
```

![Image](https://github.com/user-attachments/assets/856d7be3-d4a8-4ebf-a347-de7981ae7a24)

**Comment by Jollywatt** [comment]
Set `reflow: true` in `scale()` :)

**Comment by JL710** [comment]
Oh. I did not know that reflow exists. That works! Thanks!
